extensions/arc/src/models/postgresModel.ts

```diff
@@ -103,4 +116,87 @@ export class PostgresModel extends ResourceModel {
 			this._refreshPromise = undefined;
 		}
 	}
+
+	public async getEngineSettings(): Promise<void> {
+		await this.getConnectionProfile();
+		if (this._connectionProfile) {
+			// TODO
+			// select * from pg_settings;
+			this.engineSettingsLastUpdated = new Date();
+		}
+	}
+
+	/**
+	 * Loads the saved connection profile associated with this model. Will prompt for one if
+	 * we don't have one or can't find it (it was deleted)
+	 */
+	private async getConnectionProfile(): Promise<void> {
```

Comment:
It would be nice to have this shared with the SQL one since they share a lot of the same code - just with slightly different values for things like provider ID and messages.

Reply:
So you are saying to remove the connectionProfile being defined and instead just update the values taken from the input boxes?

Reply:
I'd probably just have this function in the base class and then define an abstract function that returns the connection.

Reply:
And then for the connection dialog maybe just have an abstract function called something like `promptForConnection` that opens the corresponding dialog and returns the connection profile from that.